### DISCUSSION TITLE
Regarding rocksdb backend for v2

### DISCUSSION THREAD

**Discussion by trickster** [xtdb/xtdb#2513] [General]
How can modify this config to enable RocksDB backend https://github.com/xtdb/xtdb/blob/2.x/docker/xtdb.edn

The old config from v1, doesn't seem to work.

```
:xtdb/tx-log {:kv-store {:xtdb/module xtdb.rocksdb/->kv-store
                        :db-dir "data/tx-log"}}
 :xtdb/document-store {:kv-store {:xtdb/module xtdb.rocksdb/->kv-store
                                :db-dir "data/docs"}}
 :xtdb/index-store {:kv-store {:xtdb/module xtdb.rocksdb/->kv-store
                             :db-dir "data/indices"}}
```

with error

```
Execution error (IllegalArgumentException) at juxt.clojars-mirrors.integrant.core/try-build-action (core.cljc:304).
No method in multimethod 'init-key' for dispatch value: :xtdb/document-store
```

**Comment by jarohen** [comment]
Hey @trickster :wave: 

The architecture for v2 is a little different - rather than the document-store and index-store we now just have an 'object-store', so RocksDB will no longer be necessary.

We're yet to write up the documentation for anything apart from in-memory use cases, but we do have an on-disk transaction-log and document-store available - you could try the following:

```clojure
(let [node-dir (io/file "/tmp/xt2")]
  {:xtdb.log/local-directory-log {:root-path (io/file node-dir "log")}
   :xtdb.object-store/file-system-object-store {:root-path (io/file node-dir "objects")}})
```

Please do bear in mind that the on-disk formats are still subject-to-change through the early access - best not to rely on backwards-compatibility or durability between snapshot versions!

Cheers,

James
